homeassistant/components/comfoconnect/__init__.py

```diff
@@ -71,20 +96,49 @@ def setup(hass, config):
     hass.data[DOMAIN] = ccb
 
     # Start connection with bridge
-    ccb.connect()
+    await ccb.connect()
 
     # Schedule disconnect on shutdown
-    def _shutdown(_event):
-        ccb.disconnect()
-
-    hass.bus.listen_once(EVENT_HOMEASSISTANT_STOP, _shutdown)
+    async def _shutdown(_event):
+        await ccb.disconnect()
 
-    # Load platforms
-    discovery.load_platform(hass, "fan", DOMAIN, {}, config)
+    hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, _shutdown)
+    entry.async_on_unload(entry.add_update_listener(options_update_listener))
 
+    for platform in PLATFORM:
+        hass.async_create_task(
+            hass.config_entries.async_forward_entry_setup(entry, platform)
+        )
```

Comment:
```suggestion
    hass.config_entries.async_setup_platforms(entry, PLATFORM)
```
I also suggest to rename the const `PLATFORM` to `PLATFORMS` to make it more clear it is a list of platforms

Reply:
Maybe this was missed, please replace the loop with `hass.config_entries.async_setup_platforms(entry, PLATFORMS)` (it does exactly what is in your loop)

Reply:
ah, sorry, I misread your suggestion!